common.mk

```diff
@@ -114,46 +114,49 @@ generator_temp: $(SCALA_SOURCES) $(sim_files) $(SCALA_BUILDTOOL_DEPS) $(EXTRA_GE
 		--top-module $(MODEL_PACKAGE).$(MODEL) \
 		--legacy-configs $(CONFIG_PACKAGE):$(CONFIG) \
 		$(EXTRA_CHISEL_OPTIONS))
+	sed -i '1d' $(ANNO_FILE)
+	echo "[" >> t.json
+	echo "{" >> t.json
+	echo "  \"class\": \"sifive.enterprise.firrtl.ModuleHierarchyAnnotation\"," >> t.json
+	echo "  \"filename\": \"./mod-he.json\"" >> t.json
+	echo "}," >> t.json
+	mv $(ANNO_FILE) t2.json
+	cat t.json t2.json > $(ANNO_FILE)
+	rm t.json t2.json
 
 .PHONY: firrtl
 firrtl: $(FIRRTL_FILE)
 
 #########################################################################################
 # create verilog files rules and variables
 #########################################################################################
-REPL_SEQ_MEM = --infer-rw --repl-seq-mem -c:$(MODEL):-o:$(TOP_SMEMS_CONF)
-HARNESS_CONF_FLAGS = -thconf $(HARNESS_SMEMS_CONF)
-
-TOP_TARGETS = $(TOP_FILE) $(TOP_SMEMS_CONF) $(TOP_ANNO) $(TOP_FIR) $(sim_top_blackboxes)
-HARNESS_TARGETS = $(HARNESS_FILE) $(HARNESS_SMEMS_CONF) $(HARNESS_ANNO) $(HARNESS_FIR) $(sim_harness_blackboxes)
+CIRCT_TARGETS = $(VSRC_SMEMS_FILE) $(VSRC_MODH_JSON)
 
 # DOC include start: FirrtlCompiler
 # NOTE: These *_temp intermediate targets will get removed in favor of make 4.3 grouped targets (&: operator)
 .INTERMEDIATE: firrtl_temp
-$(TOP_TARGETS) $(HARNESS_TARGETS): firrtl_temp
+$(CIRCT_TARGETS): firrtl_temp
 	@echo "" > /dev/null
 
 firrtl_temp: $(FIRRTL_FILE) $(ANNO_FILE) $(VLOG_SOURCES)
-	$(call run_scala_main,tapeout,barstools.tapeout.transforms.GenerateTopAndHarness,\
-		--allow-unrecognized-annotations \
-		--output-file $(TOP_FILE) \
-		--harness-o $(HARNESS_FILE) \
-		--input-file $(FIRRTL_FILE) \
-		--syn-top $(TOP) \
-		--harness-top $(VLOG_MODEL) \
-		--annotation-file $(ANNO_FILE) \
-		--top-anno-out $(TOP_ANNO) \
-		--top-dotf-out $(sim_top_blackboxes) \
-		--top-fir $(TOP_FIR) \
-		--harness-anno-out $(HARNESS_ANNO) \
-		--harness-dotf-out $(sim_harness_blackboxes) \
-		--harness-fir $(HARNESS_FIR) \
-		$(REPL_SEQ_MEM) \
-		$(HARNESS_CONF_FLAGS) \
-		--target-dir $(build_dir) \
-		--log-level $(FIRRTL_LOGLEVEL) \
-		$(EXTRA_FIRRTL_OPTIONS))
-	touch $(sim_top_blackboxes) $(sim_harness_blackboxes)
+	$(SCRATCH_HOME)/circt/build/bin/firtool \
```

Comment:
What is this?

Reply:
`firtool` is the LLVM MLIR-based CIRCT compiler. See https://github.com/llvm/circt  and https://github.com/chipsalliance/chisel3/blob/f4050157dc59cffff52eccfcd8613c9de058ae5d/src/main/scala/circt/stage/phases/CIRCT.scala#L121

Reply:
Related to your comment below - is firtool essentially some Phase that is pre-compiled into a binary (with Ninja, it seems), which is the main reason why it's faster than what we have now?

Reply:
It is basically a complete rewrite of the FIRRTL compiler passes into a LLVM binary (C++). So it is a full replacement of the scala FIRRTL compiler.